homeassistant/util/unit_conversion.py

```diff
@@ -0,0 +1,125 @@
+"""Typing Helpers for Home Assistant."""
+from __future__ import annotations
+
+from numbers import Number
+from typing import TypeVar
+
+from homeassistant.const import (
+    ENERGY_KILO_WATT_HOUR,
+    ENERGY_MEGA_WATT_HOUR,
+    ENERGY_WATT_HOUR,
+    POWER_KILO_WATT,
+    POWER_WATT,
+    PRESSURE_PA,
+    TEMP_CELSIUS,
+    UNIT_NOT_RECOGNIZED_TEMPLATE,
+    VOLUME_CUBIC_METERS,
+)
+
+from . import (
+    pressure as pressure_util,
+    temperature as temperature_util,
+    volume as volume_util,
+)
+
+_ValueT = TypeVar("_ValueT", float, None)
+
+
+class BaseUnitConverter:
+    """Define the format of a conversion utility."""
+
+    DEVICE_CLASS: str
+    NORMALIZED_UNIT: str
+    UNIT_CONVERSION: dict[str, float]
+    VALID_UNITS: tuple[str, ...]
+
+    @classmethod
+    def convert(
+        cls, value: _ValueT, from_unit: str, to_unit: str, bypass_checks: bool = False
+    ) -> _ValueT:
```

Comment:
There is an improvement here from the previous implementions, with the added support for `None` value, and the ability to bypass checks (useful if checks are anyway already run before calling the convert function).